### PR TITLE
Bug 1526147 - Add instructions for alternative to apb run

### PR DESCRIPTION
**Pull request by dymurray**

### CODE REVIEW

File: docs/developers.md [hunk 1]
```diff
@@ -509,6 +509,19 @@ $ curl -H "Authorization: Bearer $(oc whoami -t)" -k https://asb-1338-ansible-se
...
+Because of the limitations described above, it may be desired for a user to want the same functionality as `apb run` without having to rely on `apb push` being successful. This is because `apb run` implicitly performs `apb push` first before attempting to provision the application. In order to work around this, a user must first follow the above workaround to push their image onto the internal OpenShift registry. Once the image exists, you should be able to see the image with:
+```
+$ oc get images | grep <bundle_name>
+sha256:bfaa73a5e15bf90faec343c7d5f8cc4f952987afdbc3f11a24c54c037528d2ed   172.30.1.1:5000/openshift/<bundle_name>@sha256:bfaa73a5e15bf90faec343c7d5f8cc4f952987afdbc3f11a24c54c037528d2ed
+```
+
+Now in order to provision, we can use `oc run` to launch the Bundle:
+```
+$ oc new-project run-test
+$ oc run --image 172.30.1.1:5000/openshift/<bundle_name> provision-<bundle_name> provision
```
Comment (djzager) [inline docs/developers.md]: It is my understanding that if you want the APB to have sufficient permissions to do anything in the target namespace, you will need to create a serviceaccount and (cluster)rolebinding first. Then pass that serviceaccount to the `oc run` command. By default, the service account given to the pod can't create anything (or even list).

**Review by djzager — CHANGES_REQUESTED** [review]
I think you may need to add a little bit more in the case the APB wants to do anything in the target namespace.

File: docs/developers.md (hunk 1 above)
Comment (djzager) [reply docs/developers.md]: This is from the apb-test-shim...may be useful to you.

```
    oc run "$pod_name" \
        --namespace=$apb_name \
        --env="POD_NAME=$pod_name" \
        --env="POD_NAMESPACE=$apb_name" \
        --image=$apb_name \
        --image-pull-policy=Never \
        --restart=Never \
        --attach=true \
        --serviceaccount=$apb_name \
        -- $action -e namespace=$apb_name -e cluster=$CLUSTER
```

Comment (dymurray) [reply docs/developers.md]: Ahh... my test APB just did a debug :) didn't think about that. Thanks!

**Review by djzager — APPROVED** [review]
ACK

File: docs/developers.md [hunk 2]
```diff
@@ -509,6 +509,28 @@ $ curl -H "Authorization: Bearer $(oc whoami -t)" -k https://asb-1338-ansible-se
 
 You should see a list of all bootstrapped specs and one that is labeled `localregistry-<bundle_name>`. I recommend using `|grep <bundle_name>` to help find it since the output is in JSON.
 
+### Alternative to using `apb run`
+Because of the limitations described above, it may be desired for a user to want the same functionality as `apb run` without having to rely on `apb push` being successful. This is because `apb run` implicitly performs `apb push` first before attempting to provision the application. In order to work around this, a user must first follow the above workaround to push their image onto the internal OpenShift registry. Once the image exists, you should be able to see the image with:
+```
+$ oc get images | grep <bundle_name>
```
Comment (jianzhangbjz) [inline docs/developers.md:515]: @dymurray As I described in bug [1541903](https://bugzilla.redhat.com/show_bug.cgi?id=1541903), I could NOT get the pushed image by querying the `bundle_name`, I think we should modify this command, or am I missing something?

**Pull request merged**